From 778884e8ac843468b429859091489e4539671ff7 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Tue, 10 Mar 2020 09:03:33 +0000
Subject: [PATCH] Fix compiling warnings of file drivers/soc/fsl/rcpm.c

drivers/soc/fsl/rcpm.c:90:38: warning: left shift count >= width
of type [-Wshift-count-overflow]
(u32)(((u64)(reg_offset[1] << (sizeof(u32) * 8) |
^~
drivers/soc/fsl/rcpm.c:95:39: warning: left shift count >= width
of type [-Wshift-count-overflow]
(u32)(((u64)(reg_offset[1] << (sizeof(u32) * 8) |
^~
drivers/soc/fsl/rcpm.c:101:11: warning: suggest parentheses around
 assignment used as truth value [-Wparentheses]
} while (ws = wakeup_source_get_next(ws));
^~

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 drivers/soc/fsl/rcpm.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/fsl/rcpm.c b/drivers/soc/fsl/rcpm.c
index 9a29c482fc2e..73d5bbb75cbf 100644
--- a/drivers/soc/fsl/rcpm.c
+++ b/drivers/soc/fsl/rcpm.c
@@ -87,18 +87,19 @@ static int rcpm_pm_prepare(struct device *dev)
 					}
 					/* Read value from register SCFG_SPARECR8 */
 					regmap_read(scfg_addr_regmap,
-						    (u32)(((u64)(reg_offset[1] << (sizeof(u32) * 8) |
+						    (u32)(((u64)((u64)reg_offset[1] << (sizeof(u32) * 8) |
 						    reg_offset[2])) & 0xffffffff),
 						    &reg_value);
 					/* Write value to register SCFG_SPARECR8 */
 					regmap_write(scfg_addr_regmap,
-						     (u32)(((u64)(reg_offset[1] << (sizeof(u32) * 8) |
+						     (u32)(((u64)((u64)reg_offset[1] << (sizeof(u32) * 8) |
 						     reg_offset[2])) & 0xffffffff),
 						     tmp | reg_value);
 				}
 			}
 		}
-	} while (ws = wakeup_source_get_next(ws));
+		ws = wakeup_source_get_next(ws);
+	} while (ws != NULL);
 
 	wakeup_source_get_stop(idx);
 
-- 
2.17.1

